tests/datasource/fluent/test_contexts.py

```diff
@@ -68,6 +68,35 @@ def test_add_fluent_datasource_are_persisted_without_duplicates(
     assert datasource_name not in yaml_dict["datasources"]
 
 
+def test_assets_are_persisted_on_creation_and_removed_on_deletion(
+    empty_file_context: FileDataContext,
+    db_file: pathlib.Path,
+):
+    context = empty_file_context
+
+    # ensure empty initial state
+    yaml_path = pathlib.Path(context.root_directory, context.GX_YML)
+    assert yaml_path.exists()
+    assert not yaml.load(yaml_path.read_text()).get("fluent_datasources")
+
+    datasource_name = "my_datasource"
+    asset_name = "my_asset"
+
+    context.sources.add_sqlite(
+        name=datasource_name, connection_string=f"sqlite:///{db_file}"
+    ).add_query_asset(asset_name, query="SELECT *")
```

Comment:
This query is invalid, but you wouldn't find that out until trying to get batches.

Reply:
updated to be valid query